modules/pattern/pattern.service.js

```diff
@@ -18,21 +18,25 @@ class PatternsService {
   }
 
   async addPattern(data) {
+    if (await this.checkPatternExist(data)) {
+      throw new Error(PATTERN_ALREADY_EXIST);
+    }
+    return new Pattern(data).save();
+  }
+
+  async deletePattern(id) {
+    return await Pattern.findByIdAndDelete(id);
+  }
+
+  async checkPatternExist(data) {
     const pattern = await Pattern.find({
```

Comment:
Use the method `count` here to reduce the amount of data that is transferred between app and db